static/src/js/components/SpatialDisplay/__tests__/SpatialDisplay.test.js

```diff
@@ -205,406 +196,537 @@ describe('SpatialDisplay component', () => {
-          shapefileName: 'test file',
-          shapefileSize: '42 KB',
-          isErrored: {
-            type: 'upload_shape'
-          }
-        }
+        expect(screen.queryAllByText('Shape File')).toHaveLength(1)
+        expect(screen.queryAllByText('Shape File')[0]).toBeVisible()
+
+        expect(screen.queryAllByTestId('spatial-display__loading')).toHaveLength(0)
+
+        expect(screen.queryAllByText('test file')).toHaveLength(1)
+        expect(screen.queryAllByText('(42 KB)')).toHaveLength(1)
+      })
+
+      describe('when the shapefile has selected shapes', () => {
+        test('should render a hint with number of shapes selected', () => {
+          const newPolygon = '-77.04444122314453,38.99228142151045,'
+            + '-77.01992797851562,38.79166886339155,'
+            + '-76.89415168762207,38.902629947921575,'
+            + '-77.04444122314453,38.99228142151045'
+
+          setup({
+            polygonSearch: [newPolygon],
+            shapefile: {
+              shapefileName: 'test file',
+              shapefileSize: '42 KB',
+              isLoaded: true,
+              isLoading: false,
+              selectedFeatures: ['1']
+            }
+          })
+
+          expect(screen.queryAllByText('1 shape selected')).toHaveLength(1)
+        })
       })
 
-      const filterStackItem = enzymeWrapper.find(FilterStackItem)
-      expect(filterStackItem.props().error).toEqual('To use a shapefile, please upload a zip file that includes its .shp, .shx, and .dbf files.')
+      describe('when the shapefile is the wrong type', () => {
+        test('should render an upload hint with number of shapes selected', () => {
+          setup({
+            shapefile: {
+              shapefileName: 'test file',
+              shapefileSize: '42 KB',
+              isErrored: {
+                type: 'upload_shape'
+              }
+            }
+          })
+
+          expect(screen.queryAllByTestId('filter-stack-item__error')[0].innerHTML).toEqual('To use a shapefile, please upload a zip file that includes its .shp, .shx, and .dbf files.')
+        })
+      })
     })
   })
 
-  test('componentWillReceiveProps sets the state', () => {
-    const { enzymeWrapper } = setup()
+  describe('#onSpatialRemove', () => {
+    test('calls onRemoveSpatialFilter', async () => {
+      userEvent.setup()
 
-    expect(enzymeWrapper.state().pointSearch).toEqual(undefined)
-    const newPoint = '0,0'
-    enzymeWrapper.setProps({ pointSearch: [newPoint] })
-    expect(enzymeWrapper.state().pointSearch).toEqual(newPoint)
-  })
+      const { props } = setup({
+        manuallyEntering: 'marker'
+      })
 
-  describe('#onSpatialRemove', () => {
-    test('calls onRemoveSpatialFilter', () => {
-      const { enzymeWrapper, props } = setup()
+      const { onRemoveSpatialFilter } = props
 
-      enzymeWrapper.instance().onSpatialRemove()
+      const actionBtns = screen.queryAllByTestId('filter-stack-item__action-button')
```

Comment:
```suggestion
      const actionBtns = screen.getAllByRole('button')
```
I think this can actually get rid of an existing data-test-id in `/Users/eaolivar/git/earthdata-search/static/src/js/components/FilterStack/FilterStackItem.js` on line